src/components/main-view/main-view.jsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import axios from "axios";
+import Container from "react-bootstrap/Container";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+
+import { MovieCard } from "../movie-card/movie-card";
+import { MovieView } from "../movie-view/movie-view";
+import { LoginView } from "../login-view/login-veiw";
+import { RegistrationView } from "../registration-view/registration-view";
+
+export class MainView extends React.Component {
+  constructor() {
+    // Call the superclass constructor
+    // so React can initialize it
+    super();
+
+    this.state = {
+      movies: null,
+      selectedMovie: null,
+      user: null,
```

Comment:
In regards to adding registration view you can add a state property here and we can call it `showRegistration` and instantiate it `showRegistration: false`.